parcels/fieldset.py

```diff
@@ -274,33 +237,26 @@ def _parse_wildcards(cls, paths, filenames, var):
         return paths
 
     @classmethod
-    def from_netcdf(
+    def from_data(
```

Comment:
Do you think we could get rid of `from_data` altogether @fluidnumerics-joe ? Or do you see a usecase for it that would be difficult to achieve via getting the user xarray datasets themselves?

Reply:
I think it would be ok to do away with `from_data` entirely, so long as we put together docs on how to construct an xarray dataset that "plays nice" with parcels. What do you think @erikvansebille ?

Reply:
Yep, I agree that we don't need a `from_data` anymore in v4.

Reply:
> so long as we put together docs on how to construct an xarray dataset that "plays nice" with parcels

yes, 100%. I think it would be good to even add tooling for validating datasets are compliant (both in metadata and in variable naming) and giving informative error messages

Reply:
I'll get `from_data` removed here

Reply:
Resolved in [0af7de9](https://github.com/OceanParcels/Parcels/pull/1946/commits/0af7de934fdd73f9bf149d1fcbafdb6942f0af9c)